tapdb/sqlc/migrations/000020_asset_unique_key.up.sql

```diff
@@ -1,42 +1,59 @@
+-- Step 0: temporary table that sets a boolean flag (has_duplicates) based on
+-- whether duplicates exist. All UPDATE and DELETE statements below are
+-- contingent on this flag.
+CREATE TABLE tmp_duplicate_check AS
+SELECT CASE
+        WHEN EXISTS (
+                SELECT 1
+        FROM assets
+        GROUP BY genesis_id, script_key_id, amount, anchor_utxo_id
+        HAVING COUNT(*) > 1
+        )
+        THEN 1
+        ELSE 0
+END AS has_duplicates;
+
 -- Step 1: If the assets were spent, some of the duplicates might not have been
 -- updated on that flag. To make sure we can properly group on the spent flag
 -- below, we now update all assets that are spent.
 UPDATE assets
 SET spent = true
-WHERE asset_id IN (SELECT a.asset_id
-                   FROM assets a
-                            JOIN managed_utxos mu
-                                 ON a.anchor_utxo_id = mu.utxo_id
-                            JOIN chain_txns ct
-                                 ON mu.txn_id = ct.txn_id
-                            LEFT JOIN asset_transfer_inputs ati
-                                      ON ati.anchor_point = mu.outpoint
-                   WHERE a.spent = false
-                     AND ati.input_id IS NOT NULL);
+WHERE (SELECT has_duplicates
+        FROM tmp_duplicate_check) = 1
+        AND asset_id IN (SELECT a.asset_id
+        FROM assets a
+                JOIN managed_utxos mu
+                ON a.anchor_utxo_id = mu.utxo_id
+                JOIN chain_txns ct
+                ON mu.txn_id = ct.txn_id
+                LEFT JOIN asset_transfer_inputs ati
+                ON ati.anchor_point = mu.outpoint
+        WHERE a.spent = false
+                AND ati.input_id IS NOT NULL);
 
 -- Step 2: Create a temporary table to store the minimum asset_id for each
 -- unique combination.
 CREATE TABLE tmp_min_assets AS
 SELECT MIN(asset_id) AS min_asset_id,
-       genesis_id,
-       script_key_id,
-       amount,
-       anchor_utxo_id,
-       spent
+        genesis_id,
+        script_key_id,
+        amount,
+        anchor_utxo_id,
+        spent
```

Comment:
Are these spaces intended to be added? Justification updates happen in a few places

Reply:
@gijswijs would you mind splitting the commit into formatting changes and other changes please